src/control/controlproxy.h

```diff
@@ -173,24 +154,9 @@ class ControlProxy : public QObject {
     void valueChanged(double);
 
   protected slots:
-    /// Receives the value from the master control by a unique direct connection
-    void slotValueChangedDirect(double v, QObject* pSetter) {
```

Comment:
The different equal functions are there for the unique direct connection. 

When one consumer is a in the same thread and one else is in another thread, we need two jump in addresses to maintain the unique connection. This was at least required when writing the code. 

Did you check how Qt handles the situation today? If you are not certain we should keep the code as it is.

Reply:
I didn't notice the Qt::UniqueConnection flag. The documentation of this hacky code is insufficient. Unmaintainable.

Reply:
Thinking about this again, this code has been written having a single Proxy per Control in mind. 
Today we have in many cases multiple proxies one for each connection, where this hacky code is not required.  

Unfortunately it bypasses the quinine connection Flag. Two proxies in the same thread will lead to two queued up signals and two semaphores locked (if the qt code is kept the same since I have last looked into it)

Options: 
* leave this alone and revert to the single functions 
* dive into the qt code and clarify the situation 
* Introduce a less hacky solution that has another proxy layer one, one per receiving thread. 

Maybe we can utilize a QThreadStorage for this ...

Reply:
All has started with this bug: https://bugs.launchpad.net/mixxx/+bug/1406124